packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,281 @@
+
+![ ](spec-images/select-6.png)
+
+#### Icons
+
+The client-user should be able to specify whether they want left icons on menu items or not. Icons show up in menu items to the left of text and are displayed as part of the value in the field button.
+
+![ ](spec-images/select-7.png)
+
+#### Groups
+
+> QUESTION: IS THE SL "VIEW" PICKER A SELECT (SHOULD IT BE)? WHAT ABOUT GROUPS WITHOUT LABELS, WHAT ABOUT A SELECTION GROUP VS. AN ACTION GROUP OF MENU ITEMS (A.K.A MEGA-MENUS)?
+
+The client-user should be able to create groups for menu items in the menu. Groups are not interactable. Filtering by group name should show all the options under that group, whether or not it is included in the option text.
+
+![ ](spec-images/select-8.png)
+
+#### Secondary Text
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify whether they want secondary text on menu items or not.
+
+![ ](spec-images/select-9.png)
+
+The client-user should be able to specify whether they want other information (including groups) displayed as secondary text in the field input.
+
+![ ](spec-images/select-10.png)
+
+#### Selection
+
+> NOTE: NOT SURE THIS IS A REAL USE CASE FOR SELECT COMPONENT, SEE FUTURE CONSIDERATIONS FOR MULTI-SELECTION
+
+The client-user should be able to specify whether they want to display selection with or without a check mark.
+
+![ ](spec-images/select-11.png)
+
+#### Filter
+
+The client-user should be able to specify whether they want filtering available in the select component. The filter field appears in the menu.
+
+![ ](spec-images/select-12.png)
+
+### Behavior
+
+#### Minimum Width
+
+> NOTE: NEEDS DEV REVIEW
+
+The field button minimum width is three times the height of the component (to make components targetable on touch screens). The menu width is the same as the field button until reaching the minimum width (without truncating) of the longest string in the menu items.
+
+![ ](spec-images/select-13.png)
+
+#### Menu Height
+
+The menu height should be enough to display five to seven menu items before overflowing with a vertical scrollbar (not pictured).
+
+#### Overflow Text
+
+> NOTE: NEEDS DEV & VISUAL DESIGN REVIEW
+
+All text in the select component should truncate, not wrap.
+
+![ ](spec-images/select-14.png)
+
+#### Progress & Loading
+
+> NOTE: HOW THIS (AND ANY) FEATURE IS INCLUDED IN IMPLEMENTATION IS UP TO DEVELOPERS
+
+When displaying menu items in the menu lags or is expected to take longer than normal, the menu should indicate that it's loading. The "Loading…" display is not interactive.
+
+![ ](spec-images/select-15.png)
+
+> NOTE: IS THIS NEEDED? IF YES, NEEDS VISUAL DESIGN & TECH WRITER
+
+Menus with dynamically loading menu items may have a "Load more" button. Clicking the "Load more" button should resolve into the "Loading…" indication, then load more menu items and scroll the menu as needed.
```

Comment:
I saw [a proposal in SLE](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullRequest/565673#1695761223) today for a dropdown (unclear if select or combobox) that is used to pick from hundreds/thousands of items so it initially loads only the first 100 items alphabetically. In that proposal, the only way to see more items was to start typing in the filter, which would trigger a request for items that match that filter text. (aside: I'm not sure what would happen if more than 100 items match the filter text...maybe users would have to refine their filter to see them.)

Do you think that proposed workflow aligns with everything in your spec? Any updates or recommendations we should add?

The main thing I can think to change in this document (if you like that workflow) would be to clarify in the filtering section that the filter should apply to all items, even ones that are on the server and aren't currently visible in the component.

Reply:
Sorry, I meant to update this section of the spec. What Meyer and I decided on is that we will provide a general API for a slot that could potentially be used by clients to develop the interaction design provided here is they want, and we may document the progress/loading design pattern in the future, but for now it doesn't make sense to include this at the component level. The proposal you describe should work if that's what they decide they want to do.